central/telemetry/billingmetrics/data.go

```diff
@@ -0,0 +1,69 @@
+package billingmetrics
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	bmetrics "github.com/stackrox/rox/central/billingmetrics/store"
+	cluStore "github.com/stackrox/rox/central/cluster/datastore"
+	"github.com/stackrox/rox/central/sensor/service/pipeline/clustermetrics"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/protoconv"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stackrox/rox/pkg/set"
+)
+
+var (
+	clusterReader = sac.AllowFixedScopes(
+		sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+		sac.ResourceScopeKeys(resources.Cluster))
+
+	metricsWriter = sac.AllowFixedScopes(
+		sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+		sac.ResourceScopeKeys(resources.Administration))
+
+	previousMetrics = &clustermetrics.BillingMetrics{}
+)
+
+func average(metrics ...*clustermetrics.BillingMetrics) clustermetrics.BillingMetrics {
+	n := int64(len(metrics))
+	a := clustermetrics.BillingMetrics{}
+	if n == 0 {
+		return a
+	}
+	for _, m := range metrics {
+		a.TotalNodes += m.TotalNodes
+		a.TotalCores += m.TotalCores
+	}
+	a.TotalNodes /= n
+	a.TotalCores /= n
+	return a
+}
+
+func getClusterIDs() (set.StringSet, error) {
+	ctx := sac.WithGlobalAccessScopeChecker(context.Background(), clusterReader)
+
+	clusters, err := cluStore.Singleton().GetClusters(ctx)
```

Comment:
By having the store/datastore not be part of a struct and that struct as a receiver to this function, this code is not testable

Reply:
Right, fixed in #7138.